PlayChat/AppDelegate.swift

```diff
@@ -14,208 +14,216 @@
-  
+  let IBX: String = "inbox"
+  let CHS: String = "channels"
+  let REQLOG: String = "requestLogger"
+  var chanArray: [String] = []
+  var maxMessages: UInt?
+
   var window: UIWindow?
   var storyboard: UIStoryboard?
   var navigationController: UINavigationController?
-  var tabBarController: UITabBarController!
-  var msgViewController : MessageViewController?
-  
+  var tabBarController: UITabBarController?
+  var msgViewController: MessageViewController?
+
   var user: GIDGoogleUser!
   var inbox: String?
-  var ref: FIRDatabaseReference!
+  var ref: DatabaseReference!
   var msgs: [Message] = []
-  var channelViewDict: [String : UITableView] = [:]
+  var channelViewDict: [String: UITableView] = [: ]
   var fbLog: FirebaseLogger?
-  
-  func application(application: UIApplication, didFinishLaunchingWithOptions
-    launchOptions: [NSObject: AnyObject]?) -> Bool {
-    var configureError: NSError?
-    GGLContext.sharedInstance().configureWithError(&configureError)
-    assert(configureError == nil, "Config error: \(configureError)")
-    
-    let path = NSBundle.mainBundle().pathForResource("Info", ofType: "plist")!
-    let dict = NSDictionary(contentsOfFile: path) as! [String: AnyObject]
-    let channels = dict["Channels"] as! String
-    let chanArray = channels.componentsSeparatedByString(",")
-    maxMessages = dict["MaxMessages"] as! UInt
+
+  func application(_ application: UIApplication, didFinishLaunchingWithOptions
+    launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
+
+    if let path = Bundle.main.path(forResource: "Info", ofType: "plist"),
+       let dictionary = NSDictionary(contentsOfFile: path) as? [String: AnyObject] {
+      chanArray = dictionary["Channels"]!.components(separatedBy: ",")
+      maxMessages = dictionary["MaxMessages"]! as? UInt
+    } else {
+      print("Couldn't read 'Info.plist' file")
+    }
 
     storyboard = UIStoryboard(name: "Main", bundle: nil)
     navigationController = storyboard!.instantiateInitialViewController()
       as? UINavigationController
+
     tabBarController = self.storyboard!
-      .instantiateViewControllerWithIdentifier("TabBarController")
-      as! UITabBarController
-    msgViewController = MessageViewController(maxMessages: maxMessages)
+        .instantiateViewController(withIdentifier: "TabBarController")
+        as? UITabBarController
+    msgViewController = MessageViewController(maxMessages: maxMessages!)
     tabBarController?.delegate = msgViewController
-    
+
     tabBarController?.viewControllers = [buildChannelView(chanArray[0])]
-    for i in 1...chanArray.count-1 {
+    for i in 1...chanArray.count - 1 {
       tabBarController?.viewControllers?.append(buildChannelView(chanArray[i]))
     }
-    
-    FIRApp.configure()
-    GIDSignIn.sharedInstance().clientID = FIRApp.defaultApp()?.options.clientID
+
+    FirebaseApp.configure()
+
+    GIDSignIn.sharedInstance().clientID = FirebaseApp.app()?.options.clientID
```

Comment:
This should still be `FirebaseApp.defaultApp()`.

Reply:
Mmm Xcode complains with the following message:

```
'defaultApp()' was obsoleted in Swift 3 (FirebaseCore.FirebaseApp)
```
I guess this is a recommendation for earlier versions?

Reply:
Yep, I was wrong: https://firebase.google.com/docs/reference/swift/firebasecore/api/reference/Classes/FirebaseApp#/c:objc(cs)FIRApp(cm)defaultApp

This should be `app()`, as is :)